tests/std/tests/GH_002168_regex_overflow/test.cpp

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <regex>
+
+using namespace std;
+
+bool shouldThrow(const char* const regexString) {
+    try {
+        regex regex{regexString, regex_constants::ECMAScript};
+        return false;
+    } catch (const regex_error& e) {
+        return e.code() == regex_constants::error_backref;

```

Comment:
Maybe it is better to embed `assert` into test function. This way there would be two `assert`s, and this gives more immediate context when debugging.